src/Microsoft.DotNet.Interactive.Formatting/DefaultHtmlFormatterSet.cs

```diff
@@ -52,72 +55,121 @@ internal class DefaultHtmlFormatterSet
                                 values)));
 
                     view.WriteTo(writer, HtmlEncoder.Default);
+                    return true;
                 }),
 
-                new HtmlFormatter<IHtmlContent>((view, writer) => view.WriteTo(writer, HtmlEncoder.Default)),
+                new HtmlFormatter<PocketView>((context, view, writer) =>
+                {
+                    view.WriteTo(writer, HtmlEncoder.Default);
+                    return true;
+                }),
 
-                new HtmlFormatter<ReadOnlyMemory<char>>((memory, writer) =>
+                new HtmlFormatter<IHtmlContent>((context, view, writer) =>
+                {
+                    view.WriteTo(writer, HtmlEncoder.Default);
+                    return true;
+                }),
+
+                new HtmlFormatter<ReadOnlyMemory<char>>((context, memory, writer) =>
                 {
                     PocketView view = span(memory.Span.ToString());
 
                     view.WriteTo(writer, HtmlEncoder.Default);
+                    return true;
                 }),
 
-                new HtmlFormatter<string>((s, writer) => writer.Write(s.HtmlEncode())),
+                new HtmlFormatter<string>((context, s, writer) =>
+                {
+                    // If PlainTextPreformat is true, then strings
+                    // will have line breaks and white-space preserved
+                    writer.Write(HtmlFormatter.FormatString(s));
+                    return true;
+                }),
 
-                new HtmlFormatter<TimeSpan>((timespan, writer) =>
+                new HtmlFormatter<TimeSpan>((context, timespan, writer) =>
                 {
                     PocketView view = span(timespan.ToString());
                     view.WriteTo(writer, HtmlEncoder.Default);
+                    return true;
                 }),
 
-                new HtmlFormatter<Type>((type, writer) =>
+                new HtmlFormatter<Type>((context, type, writer) =>
                 {
-                    PocketView view = span(
-                        a[href: $"https://docs.microsoft.com/dotnet/api/{type.FullName}?view=netcore-3.0"](
-                            type.ToDisplayString(PlainTextFormatter.MimeType)));
+                    string text = type.ToDisplayString(PlainTextFormatter.MimeType);
+                    
+                    // This is approximate
+                    bool isKnownDocType =
+                      type.Namespace != null &&
```

Comment:
If anyone knows how to improve this approximation beyond this please chime in

Reply:
The docs.microsoft.com APIs are quite good, and there's also a NuGet index allowing search by type, so we can eventually improve the link generation quite a lot. I think the rough heuristic here is fine for now.